createFolders.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+PROJECT_NAME=data-and-algorithms
+DIRECTORY=${PWD}
+BASE_DIRECTORY=$(echo "$DIRECTORY" | rev | cut -d "/" -f1 | rev)
+
+if [[("$1" != "")]]; then
+   if [[("$BASE_DIRECTORY" != "$PROJECT_NAME")]]; then
+      echo "Make sure you are in $PROJECT_NAME project"
+      exit
+   fi
+   rootfileName=$1
```

Comment:
you should always quote all your variables. otherwise if they contain a space, they will be expanded to unexpected stuff!

```shell
bash createFolders.sh "hello world"
```

would cause this line to expand to

```shell
rootfileName=hello world
```

which would try to invoke `world` with the `rootfileName=hello` environment variable.

also, if this is the programming language, maybe call it `programmingLanguage`.